src/Symfony/Component/Yaml/Inline.php

```diff
@@ -562,6 +564,18 @@ private static function evaluateScalar($scalar, $flags, $references = array())
                         }
 
                         return;
+                    case 0 === strpos($scalar, '!php/const:'):
+                        if (self::$constantSupport) {
+                            $constant = @constant(substr($scalar, 11));
+
+                            if (null === $constant && self::$exceptionOnInvalidType) {
+                                throw new ParseException(sprintf('The constant "%s" is not defined.', substr($scalar, 11)));
+                            }
+
+                            return $constant;
+                        }
```

Comment:
This must throw an exception when constant support is disabled and `$exceptionOnInvalidType` is set (like we do for object support).